tests/app/controller/command/commands/mention_test.py

```diff
@@ -0,0 +1,49 @@
+"""Test mention command parsing."""
+from app.controller.command.commands.mention import MentionCommand
+from db import DBFacade
+from flask import Flask
+from app.model import User
+from unittest import mock, TestCase
+
+user = 'U123456789'
+user2 = 'U234567891'
+
+
+class MentionCommandTest(TestCase):
+    """Test cases for mentions."""
+
+    def setUp(self):
+        """Set up test environment."""
+        self.app = Flask(__name__)
+        self.mock_facade = mock.MagicMock(DBFacade)
+        self.testcommand = MentionCommand(self.mock_facade)
+
+    def test_handle_no_input(self):
+        """Test handle command with no additional args."""
+        self.assertEqual(self.testcommand.handle('UFJ42EU67', user),
+                         ("invalid command", 200))
+
+    def test_handle_wrong_input(self):
+        """Test handle command with unsupported function."""
+        self.assertEqual(self.testcommand.handle('UFJ42EU67 --', user),
+                         (self.testcommand.unsupported_error, 200))
+
+    def test_handle_add_karma_to_another_user(self):
+        """Test handle command with karma to another user."""
+        user = User('U123456789')
+        user.name = 'U123456789'
+        self.mock_facade.retrieve.return_value = user
+        self.assertEqual(self.testcommand.handle(('UFJ42EU67 ++'), user),
+                         ("gave 1 karma to U123456789", 200))
```

Comment:
So the user `U123456789` ran the command `/rocket UFJ42EU67 ++`, and the result is that they gave karma to themselves??? Line 35 shouldn't return the user calling, it should return the user receiving. The handle command should have thrown a `TypeError` because it should handle the `user_id` and not the user object. Same with those above.

Also, there is no need for parentheses for the first argument of `.handle`.

Reply:
Ah yes. Good catch!